### PR TITLE
Add synthetic data retriever class

### PR DESCRIPTION
**Pull request by nv-hwoo**
This PR is to extract out the synthetic data generation part into a separate API.
This PR only handles addition of the new API and does _not_ contain
 - lifting the dependencies to the endpoint (will have follow up PR)
 - integrating into the current code as it requires @debermudez's factory code to be in place.

### CODE REVIEW

File: genai-perf/genai_perf/inputs/synthetic_data_retriever.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        synthetic_dataset = []
+        for _ in range(self.config.num_prompts):
+            prompt = SyntheticPromptGenerator.create_synthetic_prompt(
+                self.config.tokenizer,
+                self.config.prompt_tokens_mean,
+                self.config.prompt_tokens_stddev,
+            )
+            data = {"text_input": prompt}
+
+            if self.config.output_format == OutputFormat.OPENAI_VISION:
```
Comment (dyastremsky) [inline genai-perf/genai_perf/inputs/synthetic_data_retriever.py:53]: Thanks for including images in your initial work! Are you going to refactor this out later? We need to have the data format be standard, so we'll need to have a way to create synthetic images for any APIs that use them.

I think we do not want any logic specific to the output formats on the retriever side when we finish modularization.

**Comment by dyastremsky** [comment]
Also, are you planning to pull the current synthetic code out in a future PR? It looks like it was not deleted in this PR.

File: genai-perf/genai_perf/inputs/synthetic_data_retriever.py (hunk 1 above)
Comment (nv-hwoo) [reply genai-perf/genai_perf/inputs/synthetic_data_retriever.py:53]: yeah sorry for not adding any explanation on the PR. This PR is just for splitting up the synthetic part into a separate API, so we will have a follow up PR for further refining.

**Comment by nv-hwoo** [comment]
@dyastremsky you are right this PR did not remove the current code. That requires input factory code to be in place so this PR only adds the new API and will be integrated into the new workflow in the follow up PRs.

File: genai-perf/genai_perf/inputs/synthetic_data_retriever.py (hunk 1 above)
Comment (dyastremsky) [reply genai-perf/genai_perf/inputs/synthetic_data_retriever.py:53]: Got it, thanks for clarifying!

**Review by dyastremsky — APPROVED** [review]

**Pull request merged**